spring-integration-ftp/src/main/java/org/springframework/integration/ftp/session/FtpSession.java

```diff
@@ -144,6 +144,9 @@ public void append(InputStream inputStream, String path) throws IOException {
 	@Override
 	public void close() {
 		try {
+			if (this.readingRaw.get()) {
```

Comment:
Yeah... That's great, but we can use `CachingSessionFactory` and its `CachedSession` has a bit different `close()` impl. 
So, we must take care there about `finalizeRaw()` as well, in case we returning back to `pool`.